ctru-rs/src/services/ac.rs

```diff
@@ -0,0 +1,349 @@
+//! The Automatic Connection (AC) service handles Wi-Fi and network settings.
+//! It can:
+//! - Connect to a network or slot
+//! - Get information about a network, such as its SSID or security settings
+use crate::error::ResultCode;
+
+/// Handle to the Automatic Connection (AC) service, that handles Wi-Fi and network settings.
+pub struct Ac(());
+
+impl Ac {
+    /// Initialize a new service handle.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "acInit")]
+    pub fn new() -> crate::Result<Ac> {
+        unsafe {
+            ResultCode(ctru_sys::acInit())?;
+            Ok(Ac(()))
+        }
+    }
+
+    /// Waits for an internet connection
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Waiting for an internet connection...");
+    /// ac.wait_internet_connection()?;
+    /// println!("Connected.");
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "acWaitInternetConnection")]
+    pub fn wait_internet_connection(&self) -> crate::Result<()> {
+        unsafe {
+            ResultCode(ctru_sys::acWaitInternetConnection())?;
+
+            Ok(())
+        }
+    }
+
+    /// Returns whether the console is connected to Wi-Fi
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Wi-Fi connected: {}", ac.get_wifi_status()?);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetWifiStatus")]
+    pub fn get_wifi_status(&self) -> crate::Result<bool> {
+        unsafe {
+            let mut ret = 0u32;
+            ResultCode(ctru_sys::ACU_GetStatus(&mut ret))?;
+
+            Ok(ret == 3)
+        }
+    }
+
+    /// Returns the [SecurityMode] of the currently connected network, or error if the console isn't connected to any network.
```

Comment:
Nit: we usually use code blocks to specify doc links.
```suggestion
    /// Returns the [`SecurityMode`] of the currently connected network, or error if the console isn't connected to any network.
```